### PR TITLE
ddl: Delete TiFlash sync status from etcd when table is truncated or dropped

### PR DESCRIPTION
**Pull request by CalvinNeo**
Signed-off-by: CalvinNeo <calvinneo1995@gmail.com>

### What problem does this PR solve?

Issue Number: close #37168

Problem Summary:

If a table with TiFlash replica is dropped of truncated, we have to delete its progress in etcd. Otherwise will cause too much obsolette keys in etcd if there are frequent dropping/truncating, which can lead to etcd oom.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lcwangchao — APPROVED** [review]

File: domain/infosync/info.go [hunk 1]
```diff
@@ -1167,3 +1167,23 @@ func DeleteInternalSession(se interface{}) {
 	}
 	sm.DeleteInternalSession(se)
 }
+
+// SetEtcdClient is only used for test
```
Comment (zimulala) [inline domain/infosync/info.go:1171]:
```suggestion
// SetEtcdClient is only used for test.
```

File: domain/infosync/info.go [hunk 2]
```diff
@@ -1167,3 +1167,23 @@ func DeleteInternalSession(se interface{}) {
...
+func SetEtcdClient(etcdCli *clientv3.Client) {
+	is, err := getGlobalInfoSyncer()
+
+	if err != nil {
+		return
+	}
+	is.etcdCli = etcdCli
+}
+
+// GetEtcdClient is only used for test
```
Comment (zimulala) [inline domain/infosync/info.go:1181]:
```suggestion
// GetEtcdClient is only used for test.
```

File: ddl/table.go [hunk 3]
```diff
@@ -629,6 +635,12 @@ func onTruncateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ erro
 		return ver, infoschema.ErrTableNotExists.GenWithStackByArgs(job.SchemaName, tblInfo.Name.O)
 	}
 
+	if tblInfo.TiFlashReplica != nil {
+		e := infosync.DeleteTiFlashTableSyncProgress(tblInfo.ID)
```
Comment (zimulala) [inline ddl/table.go:639]: Could we put this logic to https://github.com/pingcap/tidb/pull/37184/files#diff-9dd5a0976146b94bb3074ca04e8131dffc03096505755440e64760b42bbde0fdL687? 
Then we can remove line 638.

**Review by zimulala — APPROVED** [review]
LGTM

**Review by crazycs520 — APPROVED** [review]

**Pull request merged**